tests/run-make/symbols-all-mangled/rmake.rs

```diff
@@ -41,7 +41,18 @@ fn symbols_check_archive(path: &str) {
             continue; // Unfortunately LLVM doesn't allow us to mangle this symbol
         }
 
-        panic!("Unmangled symbol found: {name}");
+        if name.starts_with("anon.") && name.contains(".llvm.") {
+            // Starting in LLVM 21 we see anonymous functions from libcore that look like
+            // anon.ae38fc0c63519f015fe99b15e38f558a.2.llvm.10001621870744393187
+            continue;
+        }
+
+        if name.starts_with("_rust_extern_with_linkage___dso_handle.llvm") {
```

Comment:
I think ignoring all symbols containing `.llvm.` is fine. The hash after it exists to prevent conflicts when thinLTO makes a symbol public.

Reply:
Done!

Reply:
Forgot to push?

Reply:
arg, got burned by git's horrible ux again, actually amended and pushed this time